### PR TITLE
[docs] Capture `base64()` in docs/functions/text

### LINKED ISSUE #1084: Update - adding `base64()` function to docs
Describe the requested changes:

In the core repo QuestDB has `base64()` function that is not yet documented:

[Base64FunctionFactory.java](https://github.com/questdb/questdb/blob/master/core/src/main/java/io/questdb/griffin/engine/functions/bin/Base64FunctionFactory.java)

The task involves:
Create a new section for each function, following the [template](https://github.com/questdb/questdb.io/blob/master/docs/__guidelines/template/function.md).

Location:
[/docs/reference/](https://questdb.io/docs/reference/function/text/)

### PR DESCRIPTION
**Pull request by sudiptog81**
Adds documentation for `base64()` function along with relevant examples in the relevant section.
Tested on local development setup.

Closes #1084.

image

### CODE REVIEW

File: docs/reference/function/text.md [hunk 1]
```diff
@@ -5,8 +5,38 @@ description: Text function reference documentation.
...
+- `data` is some binary data that we wish to encode.
+- `maxLength` is the intended maximum length of the encoded string.
+
+**Return value:**
+
+Return value type is `string`.
+
+:::tip
+
+`rnd_bin` can be used to generate random binary data.
```
Comment (amyshwang) [inline docs/reference/function/text.md:27]: Maybe adding a link to /docs/reference/function/random-value-generator#rnd_bin/

File: docs/reference/function/text.md [hunk 2]
```diff
@@ -5,8 +5,38 @@ description: Text function reference documentation.
 ---
 
 This page describes the available functions to assist with performing text
-manipulation such as concatenation, case conversion, and string length
-calculation.
+manipulation such as concatenation, case conversion, string length calculation
+and encoding data.
```
Comment (amyshwang) [inline docs/reference/function/text.md:9]:
```suggestion
manipulation such as concatenation, case conversion, string length calculation,
and encoding data.
```

**Review by amyshwang — CHANGES_REQUESTED** [review]
Mainly editorial review. Also requesting review from @puzpuzpuz.

**Comment by puzpuzpuz** [comment]
`base64()` works for binary columns, so probably we need to introduce a separate section for those.

**Comment by sudiptog81** [comment]
In that case, is it better to introduce another file, say `binary.md`?

File: docs/reference/function/text.md [hunk 3]
```diff
@@ -5,8 +5,38 @@ description: Text function reference documentation.
 ---
 
 This page describes the available functions to assist with performing text
-manipulation such as concatenation, case conversion, and string length
-calculation.
+manipulation such as concatenation, case conversion, string length calculation,
+and encoding data.
```
Comment (amyshwang) [inline docs/reference/function/text.md:9]: As discussed with @puzpuzpuz, this is not the right place for the function. Please create a new section `binaries` and add `base64()` to the new page. 
The intro for `text` will need to be reverted as a result. Thank you!

**Comment by sudiptog81** [comment]
@amyshwang created a new section and added it to the sidebar

File: docs/reference/function/binary.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+- `data` is the binary data to be encoded.
+- `maxLength` is the intended maximum length of the encoded string.
+
+**Return value:**
+
+Return value type is `string`.
+
+:::tip
+
+`rnd_bin` can be used to generate random binary data. [View documentation](./random-value-generator#rnd_bin).
```
Comment (puzpuzpuz) [inline docs/reference/function/binary.md:25]: The link is incorrect. Should be:
```suggestion
`rnd_bin` can be used to generate random binary data. [View documentation](/docs/reference/function/random-value-generator/#rnd_bin).
```

**Review by puzpuzpuz — APPROVED** [review]
@sudiptog81 congrats on the contribution!

File: docs/reference/function/binary.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+sidebar_label: Binary
+description: Binary function reference documentation.
+---
+
+This page describes the available functions to assist with working with binary data.
+
+## base64
+
+`base64(data, maxLength)` encodes raw binary data using the base64 encoding into
+a string with a maximum length of defined by `maxLength`.
```
Comment (amyshwang) [inline docs/reference/function/binary.md:12]:
```suggestion
a string with a maximum length defined by `maxLength`.
```

File: docs/reference/function/binary.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+- `data` is the binary data to be encoded.
+- `maxLength` is the intended maximum length of the encoded string.
+
+**Return value:**
+
+Return value type is `string`.
+
+:::tip
+
+`rnd_bin` can be used to generate random binary data. [View documentation](/docs/reference/function/random-value-generator/#rnd_bin).
```
Comment (amyshwang) [inline docs/reference/function/binary.md:25]:
```suggestion
[`rnd_bin`](/docs/reference/function/random-value-generator/#rnd_bin) can be used to generate random binary data.
```

**Review by amyshwang — CHANGES_REQUESTED** [review]
Nearly there! I believe that `introduction.md` needs to be updated with the new section too:
https://questdb.io/docs/#functions

**Comment by sudiptog81** [comment]
> Nearly there! I believe that `introduction.md` needs to be updated with the new section too: https://questdb.io/docs/#functions

Done! :smile:

**Pull request merged**